projects/igniteui-angular/src/lib/services/overlay/overlay.ts

```diff
@@ -735,6 +723,30 @@ export class IgxOverlayService implements OnDestroy {
         }
     }
 
+    private addCloseOnEscapeListener(info: OverlayInfo) {
+        if (info.settings.closeOnEscape && !this._keyPressEventListener) {
+            this._keyPressEventListener = fromEvent(this._document, 'keydown').pipe(
+                filter((ev: KeyboardEvent) => ev.key === 'Escape' || ev.key === 'Esc')
+            ).subscribe(() => {
+                const targetOverlay = this._overlayInfos[this._overlayInfos.length - 1];
+                if (targetOverlay.settings.closeOnEscape) {
+                    this.hide(targetOverlay.id);
+                }
+            });
+        }
+    }
+
+    private removeCloseOnEscapeListener() {
+        const closingOverlaysCount =
+        this._overlayInfos
+            .filter(o => o.closeAnimationPlayer && o.closeAnimationPlayer.hasStarted())
+            .length;
+        if (this._overlayInfos.length - closingOverlaysCount === 1 && this._keyPressEventListener && !this._keyPressEventListener.closed) {
```

Comment:
Don't think the `closed` check does much if you are de-ref the subscription property anyway. Is there a way for it to be still assigned yet be closed?